api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImpl.java

```diff
@@ -106,4 +114,26 @@ public IBundleProvider searchForPatients(StringAndListParam name, StringAndListP
 		
 		return searchQuery.getQueryResults(theParams, dao, translator, searchQueryInclude);
 	}
+	
+	@Override
+	@Transactional(readOnly = true)
+	public IBundleProvider getPatientEverything(TokenAndListParam patientId) {
+		SearchParameterMap theParams = new SearchParameterMap()
+		        .addParameter(FhirConstants.EVERYTHING_SEARCH_HANDLER, new StringParam())
+		        .addParameter(FhirConstants.COMMON_SEARCH_HANDLER, FhirConstants.ID_PROPERTY, patientId);
+		
+		HashSet<Include> revIncludes = new HashSet<>();
+		
+		revIncludes.add(new Include("Observation:" + FhirConstants.INCLUDE_PATIENT_PARAM));
+		revIncludes.add(new Include("AllergyIntolerance:" + FhirConstants.INCLUDE_PATIENT_PARAM));
+		revIncludes.add(new Include("DiagnosticReport:" + FhirConstants.INCLUDE_PATIENT_PARAM));
+		revIncludes.add(new Include("Encounter:" + FhirConstants.INCLUDE_PATIENT_PARAM));
+		revIncludes.add(new Include("MedicationRequest:" + FhirConstants.INCLUDE_PATIENT_PARAM));
+		revIncludes.add(new Include("ServiceRequest:" + FhirConstants.INCLUDE_PATIENT_PARAM));
+		revIncludes.add(new Include("ProcedureRequest:" + FhirConstants.INCLUDE_PATIENT_PARAM));
```

Comment:
Can we use the `FhirConstants` constants for these too? E.g.,

```java
revIncludes.add(new Include(FhirConstants.OBSERVATION + ":" + FhirConstants.INCLUDE_PATIENT_PARAM));
```

Reply:
Yes, I'll change it to that.